CHANGES.rst

```diff
@@ -1,3 +1,589 @@
+
+- Using ``Table.to_pandas()`` on an indexed ``Table`` with masked integer values
+  now correctly construct the ``pandas.DataFrame``. [#11432]
+
+- Fixed ``Table`` HTML representation in Jupyter notebooks so that it is
+  horizontally scrollable within Visual Studio Code. This was done by wrapping
+  the ``<table>`` in a ``<div>`` element. [#11476]
+
+- Fix a bug where a string-valued ``Column`` that happened to have a ``unit``
+  attribute could not be added to a ``QTable``.  Such columns are now simply
+  kept as ``Column`` instances (with a warning). [#11585]
+
+astropy.time
+^^^^^^^^^^^^
+
+- Fix a thread-safety issue with initialization of the leap-second table
+  (which is only an issue when ERFA's built-in table is out of date). [#11234]
+
+- Fixed converting a zero-length time object from UTC to 
+  UT1 when an empty array is passed. [#11516]
+
+astropy.uncertainty
+^^^^^^^^^^^^^^^^^^^
+
+- ``Distribution`` instances can now be used as input to ``Quantity`` to
+  initialize ``QuantityDistribution``.  Hence, ``distribution * unit``
+  and ``distribution << unit`` will work too. [#11210]
+
+astropy.units
+^^^^^^^^^^^^^
+
+- Move non-astronomy units from astrophys.py to a new misc.py file. [#11142]
+
+- The physical type of ``astropy.units.mol / astropy.units.m ** 3`` is now
+  defined as molar concentration.  It was previously incorrectly defined
+  as molar volume. [#11204]
+
+- Make ufunc helper lookup thread-safe. [#11226]
+
+- Make ``Unit`` string parsing (as well as ``Angle`` parsing) thread-safe. [#11227]
+
+- Decorator ``astropy.units.decorators.quantity_input`` now only evaluates return type annotations based on ``UnitBase`` or ``FunctionUnitBase`` types.
+  Other annotations are skipped over and are not attempted to convert to the correct type. [#11506]
+
+astropy.utils
+^^^^^^^^^^^^^
+
+- Make ``lazyproperty`` and ``classdecorator`` thread-safe. This should fix a
+  number of thread safety issues. [#11224]
+
+astropy.visualization
+^^^^^^^^^^^^^^^^^^^^^
+
+- Fixed a bug that resulted in some parts of grid lines being visible when they should have been hidden. [#11380]
+
+- Fixed a bug that resulted in ``time_support()`` failing for intervals of
+  a few months if one of the ticks was the month December. [#11615]
+
+astropy.wcs
+^^^^^^^^^^^
+
+- ``fit_wcs_from_points`` now produces a WCS with integer ``NAXIXn`` values. [#10865]
+
+- Updated bundled version of ``WCSLIB`` to v7.4, fixing a bug that caused
+  the coefficients of the TPD distortion function to not be written to the
+  header. [#11260]
+
+- Fixed a bug in assigning type when converting ``colsel`` to ``numpy.ndarray``. [#11431]
+
+- Added ``WCSCOMPARE_*`` constants to the list of WCSLIB constants available/exposed through the ``astropy.wcs`` module. [#11647]
+
+- Fix a bug that caused APE 14 WCS transformations for FITS WCS with ZOPT, BETA, VELO, VOPT, or VRAD CTYPE to not work correctly. [#11781]
+
+Other Changes and Additions
```

Comment:
Other changes and additions should not have module specific entries, this section is really just for the extra stuff

Reply:
Yeah, that's weird.  I don't really understand how town crier decided that, but I will just manually fix it for now.

Reply:
towncrier doesn't decide ;), it's up to people to put the fragment in the correct location or to a reviewer/bot to check that :grimacing: 

So fragments for "Other changes and additions" should go in `docs/changes/`, and if they are put in a sub-directory they will be rendered in a sub-category.

```
❯ ls -1 docs/changes/*/*other*
docs/changes/coordinates/11069.other.rst
docs/changes/coordinates/11073.other.rst
docs/changes/coordinates/11090.other.rst
docs/changes/coordinates/11449.other.rst
docs/changes/coordinates/11485.other.rst
docs/changes/coordinates/11601.other.rst
docs/changes/coordinates/11753.other.rst
docs/changes/cosmology/11530.other.rst
docs/changes/io.ascii/11919.other.rst
docs/changes/nddata/11107.other.rst
docs/changes/nddata/11553.other.rst
docs/changes/stats/11219.other.rst
docs/changes/stats/11356.other.rst
docs/changes/utils/11490.other.rst
```

Reply:
It seems `other` was chosen a lot for changes that looked neither like a bugfix nor like a feature, but not sure where those changes would have belong before ?

Reply:
there is no such thing as not a bugfix nor a feature. Then those are refactoring and thus internal and has no place in the changelog 🤷 I suspect this really is just a side effect of not reviewing those entries/name of files such as it was the practice before (e.g. I don't think mainainers would have let an entry go through in "Other" category before, and also people didn't think it's a category as it wasn't prepopulated with module names). So, we just need to clean up now at release.

Also, @eteq, keep in mind that with towncrier there are no examples provided for contributors for how a changelog entry looks like, so there is a much large spread of quantity and quality, thus more manual cleanup is needed at release time. Which I think is OK, we minimized on contributor convenience and PR conflicts not release easiness. So, the release manager certainly need to step in a clean up much more than before (note: there was quite a bit of cleanup to he changelog before, too.)

Reply:
Hmm, this is a bigger topic, methinks - should probably discuss it with the wider infrastructure group. 

But I think it is *not* desirable to increase the release manager workload in this context. After all, this is already a bottleneck and we did all the work with the backport bot to simplify the release process by having the author fix this sort of thing. So it sounds like we're trading the work of backporting for the work of changelog cleanup. But I think we can ameliorate this by pointing contributors to some examples.  I mean, we could even have a note in the docs or the PR template or somewhere else that says "look in the changelog in these sections to see how this would render"

That said, I just went through all of the "others", and they all looked like they belonged as *features* because they were "refactor to speed up".  The feature there is really the "speed up", but it's still useful to have in the changelog.  Or put another way, those are things we've had in the changelog in the past.  But they were always "new features".

Reply:
(for 4.3 I'm just going ahead and manually doing it as per the above, but I think we want to make that simpler in the long run since it's potentially quite error prone.)

Reply:
we may want to trace a few of those "other" cases and find out how those contributors first found there way there, since that will let us see where we could have intervened to put them on the happier path

Reply:
the tradeoff is days worth of work in terms of backport, and nitpicking on wordsmithing for the changelog. I don't think the latter should be a bottleneck, but of course would be nice if maintainers were picky at the time of PR opening to catch issues. But I don't think it's a have a cake and eat is situation, without rendering and seeing (not just listing in the dev docs) of good, compact examples people will just write long fluffy sentences that are not following those guidelines. 

Again, with towncrier we traded the conflicting PRs due to the changelog, the task to figure out the right version number section etc for someone to have a final look and cleanup at the end. It's a good tradeoff, could be better if maintainers are more nitpicky, but not a catastrophic situation to be in either.

Reply:
Agreed, the drawbacks we see here are a bit annoying but that remains a minor annoyance compared to the benefits. Now I guess the question is how to improve this for future releases, it would be great if we could fix the bot, also maybe we can add some checks with `pre-commit`.

Reply:
Right, to be clear I was not arguing against towncrier, just that increasing the workload on the release managers was one of the things we were trying to not do.  But I agree there's a have cake and eat it too solution, just have to decide which!

Reply:
Yep. Just to mention it here,  #11945 adds a `pre-commit` that would check those towncrier filenames and avoid this issue in the future.